homeassistant/components/permobil/config_flow.py

```diff
@@ -0,0 +1,182 @@
+            try:
+                self.p_api.set_email(user_input[CONF_EMAIL])
+            except MyPermobilClientException:
+                _LOGGER.exception("Error validating email")
+                errors["base"] = "invalid_email"
+
+            self.data.update(user_input)
+
+            await self.async_set_unique_id(self.data[CONF_EMAIL])
+            self._abort_if_unique_id_configured()
+
+        if errors or user_input is None:
+            return self.async_show_form(
+                step_id="user", data_schema=GET_EMAIL_SCHEMA, errors=errors
+            )
+
+        return await self.async_step_region()
+
+    async def async_step_region(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Invoke when a user initiates a flow via the user interface."""
+        errors: dict[str, str] = {}
+        if user_input is None:
+            # fetch the list of regions names and urls from the api
+            # for the user to select from.
+            try:
+                self.region_names = await self.p_api.request_region_names()
+                _LOGGER.debug(
+                    "region names %s",
+                    ",".join(list(self.region_names.keys())),
+                )
+            except MyPermobilAPIException:
+                _LOGGER.exception("Error requesting regions")
+                errors["base"] = "region_fetch_error"
+
+        else:
+            region_url = self.region_names[user_input[CONF_REGION]]
+
+            self.data[CONF_REGION] = region_url
+            self.p_api.set_region(region_url)
+            _LOGGER.debug("region %s", self.p_api.region)
+            try:
+                # tell backend to send code to the users email
+                await self.p_api.request_application_code()
+            except MyPermobilAPIException:
+                _LOGGER.exception("Error requesting code")
+                errors["base"] = "code_request_error"
+
+        if errors or user_input is None:
+            # the error could either be that the fetch region did not pass
+            # or that the request application code failed
+            schema = vol.Schema(
+                {
+                    vol.Required(CONF_REGION): selector.SelectSelector(
+                        selector.SelectSelectorConfig(
+                            options=list(self.region_names.keys()),
+                            mode=selector.SelectSelectorMode.DROPDOWN,
+                        )
+                    ),
+                }
+            )
+            return self.async_show_form(
+                step_id="region", data_schema=schema, errors=errors
+            )
+
+        return await self.async_step_email_code()
+
+    async def async_step_email_code(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Second step in config flow to enter the email code."""
+        errors: dict[str, str] = {}
+
+        try:
+            if user_input is not None:
```

Comment:
```suggestion
            if user_input:
```

Also move the the if condition outside of the try block